File: full-stack-asset-transfer-guide/applications/trader-typescript/src/commands/getAllAssets.ts
/*
 * Copyright IBM Corp. All Rights Reserved.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { Gateway } from '@hyperledger/fabric-gateway';
import { CHAINCODE_NAME, CHANNEL_NAME } from '../config';
import { AssetTransfer } from '../contract';

export default async function main(gateway: Gateway): Promise<void> {
    const network = gateway.getNetwork(CHANNEL_NAME);
    const contract = network.getContract(CHAINCODE_NAME);

    const smartContract = new AssetTransfer(contract);
    const assets = await smartContract.getAllAssets();

    const assetsJson = JSON.stringify(assets, undefined, 2);
    assetsJson.split('\n').forEach(line => console.log(line)); // Write line-by-line to avoid truncation
}

File: full-stack-asset-transfer-guide/applications/trader-go/commands/getAllAssets.go
package commands

import (
	"encoding/json"
	"fmt"
	"strings"

	"github.com/hyperledger/fabric-gateway/pkg/client"
)

// cmdGetAllAssets queries and prints all assets currently on the ledger.
func cmdGetAllAssets(gw *client.Gateway, _ []string) error {
	network := gw.GetNetwork(channelName())
	contract := network.GetContract(chaincodeName())

	smartContract := NewAssetTransfer(contract)
	assets, err := smartContract.GetAllAssets()
	if err != nil {
		return err
	}

	data, err := json.MarshalIndent(assets, "", "  ")
	if err != nil {
		return fmt.Errorf("failed to marshal assets: %w", err)
	}

	for _, line := range strings.Split(string(data), "\n") {
		fmt.Println(line)
	}
	return nil
}
